File: Data Structures & Algorithms/merge-intervals/submission-2.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort(key=lambda x: x[0])
        res = []
        current = intervals[0]
        for next in intervals[1:]:
            if next[0] <= current[1]:
                current[0] = min(current[0], next[0])
                current[1] = max(current[1], next[1])
            else:
                res.append(current)
                current = next

        res.append(current)
        return res

File: Data Structures & Algorithms/max-water-container/submission-2.cs
public class Solution {
    public int MaxArea(int[] heights) {
        int left = 0, right = heights.Length-1;
        int res = 0;

        while (left < right){
            int h = Math.Min(heights[left], heights[right]);
            int w = right-left;
            res = Math.Max(res, h*w);

            if (heights[left] < heights[right]) left++;
            else right--;
        }

        return res;
    }
}
